pkg/trustedresources/verify.go

```diff
@@ -78,19 +80,19 @@ func VerifyTask(ctx context.Context, taskObj *v1beta1.Task, k8s kubernetes.Inter
 		if errors.Is(err, ErrNoMatchedPolicies) {
 			switch config.GetVerificationNoMatchPolicy(ctx) {
 			case config.IgnoreNoMatchPolicy:
-				return nil
+				return &VerificationResult{VerificationResultType: VerificationSkip}
 			case config.WarnNoMatchPolicy:
 				logger := logging.FromContext(ctx)
 				logger.Warnf("failed to get matched policies: %v", err)
-				return nil
+				return &VerificationResult{VerificationResultType: VerificationWarn, Err: ErrNoMatchedPolicies}
 			}
 		}
-		return fmt.Errorf("failed to get matched policies: %w", err)
+		return &VerificationResult{VerificationResultType: VerificationError, Err: fmt.Errorf("failed to get matched policies: %w", err)}
```

Comment:
is this different from ErrNoMatchedPolicies?

Reply:
this err could be  ErrNoMatchedPolicies or other errors from getMatchedPolicies, e.g. ErrRegexMatch

Reply:
Oh sorry, I see the issue, I should return a wrapped error for warning as well, we do have information wrapped in the err and shouldn't return a new err directly

Reply:
updated